httpc_test.go

```
@@ -255,3 +256,314 @@ func TestCalculateExponentialBackoffWithJitterStillHonorsMaxDelay(t *testing.T)
+	}
+}
+
+func TestRedirectMaxRedirectsExceeded(t *testing.T) {
+	redirector := httptest.NewServer(nil)
+	redirector.Config.Handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Location", redirector.URL)
+		w.WriteHeader(http.StatusFound)
+	})
+	defer redirector.Close()
+
+	_, err := New(WithMaxRedirects(2)).GET(redirector.URL).Execute()
```

golangci-lint (2.11.4)

[error] response body must be closed

(bodyclose)